pkg/registry/registry.go

```diff
@@ -103,16 +112,20 @@ func (c *Controller) List(ctx context.Context, options ListOptions) (*api.Regist
 		return nil, err
 	}
 
-	filterArgs := filters.NewArgs()
-	filterArgs.Add("ancestor", registryImageRef)
-
-	containers, err := c.dockerClient.ContainerList(ctx, types.ContainerListOptions{
-		Filters: filterArgs,
-		All:     true,
-	})
+	containers, err := c.listContainers(ctx, filters.NewArgs(
+		filters.Arg("label", fmt.Sprintf("%s=registry", docker.ContainerLabelRole))))
 	if err != nil {
 		return nil, err
 	}
+	if len(containers) == 0 {
+		// ctlptl did not always label its registries, so fallback to using the
```

Comment:
hmmmm...i think this if branch is going to lead to really weird behavior (e.g., you delete a registry, and new registries appear).

The reason i originally did this with ancestor queries is because i wanted to be able to support registries that were created by other tools (e.g., k3d), which you wouldn't expect to have our special label.

we should probably do both listContainer() queries, and then merge the results.

Reply:
Ahh that's interesting context re: `ancestor` being intentional to explicitly support interop with k3d & friends!

I'll swap to always doing both queries - I actually started down that path originally (you might notice the unique `ContainerID`s in the tests now 😉)